src/main/java/az/edu/turing/sql_practice/social_media.sql

```diff
@@ -0,0 +1,155 @@
+                       email VARCHAR(255) UNIQUE NOT NULL CHECK (email like '%_@_.%_'),
+                       birth DATE CHECK (birth <= CURRENT_DATE),
+                       status VARCHAR(20) DEFAULT 'active',
+                       role VARCHAR(50) DEFAULT 'user',
+                       createdBy VARCHAR(255) NOT NULL DEFAULT current_user,
+                       updatedBy VARCHAR(255) NOT NULL DEFAULT current_user,
+                       createdAt TIMESTAMP NOT NULL DEFAULT NOW(),
+                       updatedAt TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+select * from users;
+
+
+INSERT INTO users (user_name, passwordd, email, birth)
+VALUES ('Test 2', '0000', 'usertest2@gmail.com', '2000-12-12');
+
+--ALTER TABLE users ALTER COLUMN email VARCHAR(255) UNIQUE NOT NULL CHECK (email ~* '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$')
+
+ALTER TABLE users
+DROP CONSTRAINT IF EXISTS users_email_check;
+
+ALTER TABLE users
+    ADD CONSTRAINT users_email_check
+        CHECK (email ~* '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$');
+
+--ALTER TABLE users DROP CONSTRAINT IF EXISTS users_createdBy_default;
+
+--ALTER TABLE users ADD CONSTRAINT users_createdBy_check DEFAULT 'user'
+
+ALTER TABLE users
+    ALTER COLUMN createdBy DROP DEFAULT;
+
+ALTER TABLE users
+    ALTER COLUMN createdBy SET DEFAULT 'user';
+
+ALTER TABLE users
+    ALTER COLUMN updatedBy DROP DEFAULT;
+
+ALTER TABLE users
+    ALTER COLUMN updatedBy SET DEFAULT 'user';
+
+UPDATE users
+SET createdBy = 'user', updatedBy = 'user';
+
+CREATE TABLE post (
+                      id bigserial PRIMARY KEY,
+                      user_id bigserial NOT NULL,
+                      title VARCHAR(255) NOT NULL,
+                      text TEXT NOT NULL,
+                      status VARCHAR(20) DEFAULT 'active',
+                      likes_count integer DEFAULT 0,
+                      comments_count integer DEFAULT 0,
+                      createdAt TIMESTAMP NOT NULL DEFAULT NOW(),
+                      updatedAt TIMESTAMP NOT NULL DEFAULT NOW(),
+                      FOREIGN KEY (user_id) REFERENCES users(id)
+);
+
+
+INSERT INTO post (user_id,title,text)
+VALUES (5,'Post 1', 'This is my new post');
+
+select * from post;
+
+CREATE TABLE comment_post (
+                              id bigserial PRIMARY KEY,
+                              user_id bigserial NOT NULL,
+                              post_id bigserial NOT NULL,
+                              text TEXT NOT NULL,
+                              status VARCHAR(20) DEFAULT 'active',
+                              likes_count integer DEFAULT 0,
+                              createdAt TIMESTAMP NOT NULL DEFAULT NOW(),
+                              updatedAt TIMESTAMP NOT NULL DEFAULT NOW(),
+                              FOREIGN KEY (user_id) REFERENCES users(id),
+                              FOREIGN KEY (post_id) REFERENCES post(id)
+);
+
+CREATE TABLE comment_comment (
```

Comment:
what is that?)

Reply:
post-a atilan commentlerle commente atilan commentleri ayri yazmisam. yeni comment-e reply vermekdir bu

Reply:
no need to store it like other table, instead you can add parent_comment_id to you comments table